### PR TITLE
Implement Element::set_url_attribute

### LINKED ISSUE #22568: Implement Element::set_url_attribute
`Element::set_url_attribute` the counterpart to [Element::get_url_attribute] is not implemented yet. Once implemented the macro [make_url_setter] should use the new method instead of `Element::set_string_attribute`.

Related to: https://github.com/servo/servo/pull/22563#discussion_r244288739 and https://github.com/servo/servo/issues/22454

[Element::get_url_attribute]: https://github.com/servo/servo/blob/master/components/script/dom/element.rs#L1487-L1500
[make_url_setter]: https://github.com/servo/servo/blob/master/components/script/dom/macros.rs#L108-L119

### PR DESCRIPTION
**Pull request by shanavas786**
---

- [ ] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #22568

- [ ] There are tests for these changes 

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/22682)

### CODE REVIEW

**Comment by shanavas786** [comment]
I am not sure if this is the correct way to do.

**Comment by nox** [comment]
I am not sure this is correct, cf. https://github.com/servo/servo/issues/22568#issuecomment-453981729.

**Comment by shanavas786** [comment]
@nox updated to parse url in getter.

what do you say on `make_URL_setter` ?

File: components/script/dom/macros.rs [hunk 1]
```diff
@@ -112,8 +112,8 @@ macro_rules! make_url_setter(
             use crate::dom::bindings::inheritance::Castable;
             use crate::dom::element::Element;
             let element = self.upcast::<Element>();
-            element.set_string_attribute(&local_name!($htmlname),
-                                         DOMString::from(value.0));
+            element.set_url_attribute(&local_name!($htmlname),
+                                         value);
```
Comment (nox) [inline components/script/dom/macros.rs]: Pretty sure tidy will complain about this, run `./mach fmt`.

Comment (nox) [reply components/script/dom/macros.rs]: Never mind, I forgot that it is a macro.

**Comment by nox** [comment]
I guess that's an ok change, I didn't realise you didn't parse the URL at set-time.

@bors-servo r+

**Pull request merged**